### PR TITLE
Add Concurrent::Async instrumentation

### PR DESCRIPTION
**Pull request by Drowze**
**What does this PR do?**
Add [Concurrent::Async](https://ruby-concurrency.github.io/concurrent-ruby/1.2.3/Concurrent/Async.html) instrumentation. 

**Motivation:**
Looking at the existing implementation for `Concurrent::Future`, it wasn't hard to replicate it to `Concurrent::Async` as well (which I use in a project) 😄 

**Additional Notes:**
None.

**How to test the change?**
Added some tests and documentation to the README.

**For Datadog employees:**
- [ ] If this PR touches code that signs or publishes builds or packages, or handles
credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -584,8 +584,8 @@ Aws::S3::Client.new.list_buckets
 
 ### Concurrent Ruby
 
-The Concurrent Ruby integration adds support for context propagation when using `::Concurrent::Future`.
-Making sure that code traced within the `Future#execute` will have correct parent set.
+The Concurrent Ruby integration adds support for context propagation when using `::Concurrent::Future` and `Concurrent::Async`,
+making sure that code traced within the `Future#execute` and `Concurrent::Async#async` will have correct parent set.
```
Comment (alai97) [inline docs/GettingStarted.md:588]:
```suggestion
The Concurrent Ruby integration adds support for context propagation when using `::Concurrent::Future` and `Concurrent::Async`, and ensures that code traced within the `Future#execute` and `Concurrent::Async#async` will have the correct parent set.
```

**Review by alai97 — APPROVED** [review]
Looks good, one wording suggestion!

**Comment by Drowze** [comment]
Good one, thank you!

**Review by marcotc — APPROVED** [review]

**Comment by marcotc** [comment]
Thank you, @Drowze!

**Comment by ivoanjo** [comment]
Hey @Drowze, 1.21.0 has just been released with this instrumentation :tada:! Give it a try :+1:

**Pull request merged**